km51/Ostapenko_Kostiantyn/laboratory1.sql

```diff
@@ -1,2 +1,132 @@
--- LABORATORY WORK 1
--- BY Ostapenko_Kostiantyn
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+змінювати структуру таблиць та створювати таблиці.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+CREATE USER ostapenko
+IDENTIFIED by ostapenko
+DEFAULT TABLESPACE "USERS"
+TEMPORARY TABLESPACE "TEMP";
+
+GRANT conect to "ostapenko";
+GRANT CREATE ANY table TO ostapenko;
+GRANT ALTER any table TO  ostapenko;
+
+
+
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+Студент пише нотатки у блокноті.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+CREATe TABLE student 
+(
+id_student varchar2(8) not null,
+name_student varchar2(30) not null
+);
+
+CREATE TABLE notes
+(
+id_notes varchar2(8) not null,
+header_notes char(30) not null
+)
+
+CREATE table notebook
+(
+id_notebook varchar2(8) not null
+
+)
+
+ALTER TABLE student add CONSTRAINT pk_name_students PRIMARY KEY (name_student);
+ALTER TABLE notebook add CONSTRAINT pk_notebook PRIMARY KEY (id_notebook);
+ALTER TABLE notes ADD CONSTRAINT fk_notes FOREIGN KEY (id_notebook) REFERENCES notebook(id_notebook);
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+
+  
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+3.a. 
+Як звуть покупця, що купив не найдорожчий товар.
+Виконати завдання в Алгебрі Кодда. 
+4 бали
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
```

Comment:
Sql : 
SELECT cust_name
FROM CUSTOMERS, ORDERS, ORDERITEMS, PRODUCTS
Where CUSTOMERS.cust_id = ORDERS.cust_id
    and orders.order_num = orderitems.order_num
    and orderitems.prod_id = products.prod_id
    and products.prod_price != max(products.prod_price)